sdk/core/azure-core/src/datetime.cpp

```diff
@@ -50,6 +52,9 @@ struct ComputeYearResult
   int SecondsLeftThisYear;
 };
 
+constexpr auto MinYear = 1601;
+constexpr auto MaxYear = 9999;
+
 constexpr int SecondsInMinute = 60;
 constexpr int SecondsInHour = SecondsInMinute * 60;
 constexpr int SecondsInDay = 24 * SecondsInHour;
```

Comment:
Do not have symbols for days; not all days have 24 hours in them.
Do we need these symbols at all?

Reply:
These are private, not visible to the outside. They are used by the cpprest's implementation which we derived ours from. There will likely be a time when we rewrite it, but I would not touch them now, cpprest works as we need it to. Good thing we have tests, so we later can make these modifications. In cpprest's logic, days are 24 hours, and it works correctly.

I think it is okay to assume that days have 24 hours in limited contexts. Days don't have 24 hours in DST-affected time, but we store time as UTC. Same goes for leap seconds - this class can parse them, but does not store them. In a context of this class being able to represent calendar dates for sending them over the wire, it works. This class operates on calendar dates, durations are not meant to be precisely accurate in the terms of physical time durations. Nevertheless, it works accurately for the time zones/DST, and leap years. It does not work timewise accurately for leap seconds, but calendar-wise, it is accurate. It is also probably not perfectly accurate from historical point either, but  for representing the datetimes in ISO/RFC formats, for information exchange, it is good implementation. For instance, in ISO format, you can create a date that is prior to the date when Gregorian calendar was even created, not to mention rules for leap days.